src/OrchardCore.Modules/OrchardCore.Media/Services/MediaFileStoreResolverMiddleware.cs

```diff
@@ -107,7 +108,11 @@ public async Task Invoke(HttpContext context)
             }, LazyThreadSafetyMode.ExecutionAndPublication)).Value;
 
             // Always call next, this middleware always passes.
-            await _next(context);
+            if (await _mediaFileStoreCache.IsCachedAsync(subPathValue))
```

Comment:
I don't think we should do this, but fix the issue in ImageSharp for over logging.

Here the request would stall, because it doesn't call `_next`

Reply:
At some point with @Skrypt we were returning a 404 through the Response, after checking that the response has not yet started.

> Here the request would stall, because it doesn't call _next

For me it make sense to do this, it would just skip the following middleware including ISharp, returning back to the "parent" / calling ones.

Reply:
Hmm, but we want the following middleware to be called, it maybe `ImageSharp` or it maybe `StaticFileMiddleware`, but we want them to be called, so that they can process the not found (and fix ImageSharp not to over log the not found).

Here we're just making a repeat of the `IsCachedAsync` that is already called earlier in the middleware (and managed with the concurrent dictionary)

So it depends what issue @Skrypt is trying to fix with this, whether it's a zero length issue, or another ?

Just need to understand the purpose behind it.

Reply:
Ok I think we all agree this is a patch in the Middleware because of ImageSharp logging level throwing an exception when it doesn't find the file we want it to process. Here, I kept it because, this cause a security issue where someone could fill the logs of a website by requesting a url that would try to resize a file that doesn't exist. We may add a comment here in the mean time that this should be removed. We can't rely on the fact that this will be fixed or not in ImageSharp.

Reply:
@deanmarcussen 

I think it's okay as we know here that it is a media file, otherwise we call the next middleware above line 60.

Unless maybe for a "media" file that is not intended to be processed by IS, and, as i remember, to be served by a subsequent custom static file provider that anyway will not find it, in fact the `DefaultMediaFileStoreCacheFileProvider` itself that is an `IMediaFileProvider` being also an `IStaticFileProvider` to serve "media" files coming from a blob and physically cached locally, but not intended to be processed by IS , but i may be wrong, as i remember.

> Here we're just making a repeat of the IsCachedAsync that is already called earlier in the middleware (and managed with the concurrent dictionary)

But the file may have been deleted right after, and when the file has been cached we don't reach again this code path.

> I'm so annoying I always want to understand why!

me too ;)

@Skrypt 

> Because, right now, we can't rely on the fact that this will be fixed or not in ImageSharp.

Yes, for now better to do something

Reply:
Ok, so if we're trying things, i.e. some extra logging, I'd rather not make other changes at the same time.

James has already agreed to the suggestion to reduce the logging, so it's just a matter of doing that being done.

I don't want to introduce differences here, between media used with blob storage vs media used without blob storage.

If we feel there's a need to address a potential logging issue, we can either do that in ImageSharp, or it would need to be a different piece of middleware that handles the issue for both normal storage and blob storage (much prefer just to handle it in ImageSharp)

Reply:
Okay, not enough focused on this PR, so just some feedbacks related to what a given comment was talking about at a given time. That said, not following the pipeline is not incompatible with the 0 length checking that is done before and anyway will throw. But okay, for testing maybe better to just add the 0 lentgh checking first.

> I don't want to introduce differences here, between media used with blob storage vs media used without blob storage.

Okay, for a local storage we would need another middleware just to check if a given media file exists, funny ;)